src/libraries/System.Private.CoreLib/src/System/IO/StringReader.cs

```diff
@@ -175,38 +172,37 @@ public override string ReadToEnd()
         //
         public override string? ReadLine()
         {
-            if (_s == null)
+            string? s = _s;
+            if (s == null)
             {
-                throw new ObjectDisposedException(null, SR.ObjectDisposed_ReaderClosed);
+                ThrowHelper.ThrowObjectDisposedException_ReaderClosed();
             }
+            if ((uint)_pos >= (uint)s.Length)
```

Comment:
I don't think this casting will help avoid bounds checks as long as the field _pos is used rather than a local.